### PR TITLE
Documents type() for image-set()

### LINKED ISSUE #4175: Issue with "image-set()": (no reference to types())
MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/image-set()

#### What information was incorrect, unhelpful, or incomplete?

The image-set() article makes no reference to the `type()` function mentioned in the spec: [title](https://drafts.csswg.org/css-images-4/#image-set-notation)https://drafts.csswg.org/css-images-4/#image-set-notation used for fallback image mime types

#### What did you expect to see?

References to the use of `type()` for mime type fall-backs in addition to the current information on uses for `resolution` (some language might just be copy pasted from similar references in the `<picture>` article). An example of using `image-set()` for mime type fall backs. A separate compatibility table for the types() function. 

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/css/image-set()`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/image-set()
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/image-set()/index.html
* Last commit: https://github.com/mdn/content/commit/a74d23a7d82ae5b3414f2a6a664e9008a909d905
* Document last modified: 2021-04-09T14:23:35.000Z

</details>

### PR DESCRIPTION
**Pull request by rachelandrew**
Working on #4306 

Fixes: #4175 

Documented the `type()` functionality, adds an example, adds to release notes. 

I need to update the data for the syntax, and also BCD.

### CODE REVIEW

File: files/en-us/web/css/image-set()/index.html [hunk 1]
```diff
@@ -26,17 +26,30 @@ <h2 id="Syntax">Syntax</h2>
...
+<p>This example shows how to use <code><a class="css" href="https://drafts.csswg.org/css-images-4/#funcdef-image-set" id="ref-for-funcdef-image-set⑨">image-set()</a></code> to provide two alternative {{cssxref("background-image")}} options, chosen depending on the resolution needed: a normal version and a high-resolution version.</p>
+
 <p>{{EmbedGHLiveSample("css-examples/images/image-set.html", '100%', 600)}}</p>
 
-<p>This example shows how to use <code><a class="css" href="https://drafts.csswg.org/css-images-4/#funcdef-image-set" id="ref-for-funcdef-image-set⑨">image-set()</a></code> to provide two alternative {{cssxref("background-image")}} options, chosen depending on the resolution needed: a normal version and a high-resolution version.</p>
+<h3 id="Using_image-set_to_provide_alternative_formats">Using image-set() to provide alternative image formats</h3>
+
+<p>In the next example the <code>type()</code> function is used to serve the image in avif and jpeg formats. If the browser supports avif, it will choose that version. Otherwise it will use the jpeg version.</p>
+
+<p>{{EmbedGHLiveSample("css-examples/images/image-set-type.html", '100%', 600)}}</p>
```
Comment (chrisdavidmills) [inline files/en-us/web/css/image-set()/index.html:52]: This example is not displaying.

File: files/en-us/web/css/image-set()/index.html [hunk 2]
```diff
@@ -26,17 +26,30 @@ <h2 id="Syntax">Syntax</h2>
...
 
+<p>This example shows how to use <code><a class="css" href="https://drafts.csswg.org/css-images-4/#funcdef-image-set" id="ref-for-funcdef-image-set⑨">image-set()</a></code> to provide two alternative {{cssxref("background-image")}} options, chosen depending on the resolution needed: a normal version and a high-resolution version.</p>
+
 <p>{{EmbedGHLiveSample("css-examples/images/image-set.html", '100%', 600)}}</p>
 
-<p>This example shows how to use <code><a class="css" href="https://drafts.csswg.org/css-images-4/#funcdef-image-set" id="ref-for-funcdef-image-set⑨">image-set()</a></code> to provide two alternative {{cssxref("background-image")}} options, chosen depending on the resolution needed: a normal version and a high-resolution version.</p>
+<h3 id="Using_image-set_to_provide_alternative_formats">Using image-set() to provide alternative image formats</h3>
+
+<p>In the next example the <code>type()</code> function is used to serve the image in avif and jpeg formats. If the browser supports avif, it will choose that version. Otherwise it will use the jpeg version.</p>
```
Comment (chrisdavidmills) [inline files/en-us/web/css/image-set()/index.html:50]: IT would be nice to capitalize the formats, i.e. JPEG, AVIF.

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Looking mostly good @rachelandrew ; just a couple of points to address.

**Comment by jsnkuhn** [comment]
possibly include a 3rd example showing type() and resolution used together? Something like for the 2x option use AVIF if available to mitigate extra filesize. If no AVIF just use 2x JPEG.

File: files/en-us/web/css/image-set()/index.html (hunk 1 above)
Comment (rachelandrew) [reply files/en-us/web/css/image-set()/index.html:52]: are you in a browser that supports type?

**Comment by rachelandrew** [comment]
> possibly include a 3rd example showing type() and resolution used together? Something like for the 2x option use AVIF if available to mitigate extra filesize. If no AVIF just use 2x JPEG.

It's a bit soon to do that due to compat issues, once this is supported in more browsers then we can add more examples.

**Comment by jsnkuhn** [comment]
should there be some mention of the lack of a natural fall back for image-set()? IE html <picture> element falls back to the  child tag if <picture> not supported. As far as i'm aware there is no equivalent in `image-set()` so a separate `background-image` declaration is needed for no `image-set()` support fall back.

**Review by chrisdavidmills — APPROVED** [review]
OK, I think I'm just going to get this merged now.

**Comment by chrisdavidmills** [comment]
> should there be some mention of the lack of a natural fall back for `image-set()`? IE html `<picture>` element falls back to the `` child tag if `<picture>` is not supported. As far as i'm aware there is no equivalent in `image-set()` so a separate `background-image` declaration is needed for no `image-set()` support fall back.

I've pushed this out to a separate issue, so it doesn't get lost.

**Pull request merged**